rfcs/12-primitive-shapes.md

```diff
@@ -0,0 +1,545 @@
+struct WedgeCollider {
+  wedge: Wedge,
+  translation: Vec3,
+  rotation: Quat,
+}
+impl Meshable for WedgeCollider {}
+impl Collider for WedgeCollider {}
+*/
+
+// Other Types
+
+struct Torus {
+  major_radius: f32,
+  tube_radius: f32,
+}
+impl Meshable for Torus {}
+
+// A 3d frustum used to represent the volume rendered by a camera, defined by the 6 planes that set the frustum limits.
+struct Frustum {
+  near: Plane,
+  far: Plane,
+  top: Plane,
+  bottom: Plane,
+  left: Plane,
+  right: Plane,
+}
+impl Meshable for Frustum {}
+
+```
+
+### 2D Geometry Types
+
+These types only exist in 2d space: their dimensions and location are only defined in `x` and `y` unlike their 3d counterparts. These types are suffixed with "2d" to disambiguate from the 3d types in user code, guide users to using 3d types by default, and remove the need for name-spacing the 2d and 3d types when used in the same scope.
+
+```rust
+struct Point2d(Vec2)
+
+struct Direction2d(Vec2)
+impl Meshable for Direction2d {}
+
+struct Ray2d {
+  point: Point2d, 
+  direction: Direction2d
+}
+impl Meshable for Ray2d {}
+
+struct Line2d {
+  point: Point2d, 
+  direction: Direction2d,
+}
+impl Meshable for Line2d {}
+
+struct LineSegment2d { 
+  start: Point2d, 
+  end: Point2d,
+}
+impl Meshable for LineSegment2d {}
+
+struct PolyLine2d<const N: usize>{
+  points: [Point2d; N],
+}
+impl Meshable for PolyLine2d {}
+
+struct Triangle2d([Point2d; 3]);
+impl Meshable for Triangle2d {}
+
+struct Quad2d([Point2d; 4]);
+impl Meshable for Quad2d {}
+
+/// A regular polygon, such as a square or hexagon.
+struct RegularPolygon2d {
+  /// The circumcircle that all points of the regular polygon lie on.
+  circumcircle: Circle2d,
+  /// Number of faces.
+  faces: u8,
+  /// Clockwise rotation of the polygon about the origin. At zero rotation, a point will always be located at the 12 o'clock position.
+  orientation: Angle,
+}
+impl Meshable for RegularPolygon2d {}
+ 
+struct Polygon2d <const N: usize>{
+  points: [Point; N],
+}
+impl Meshable for Polygon2d {}
+
+/// Circle types
+
+struct Circle2d {
+  radius: f32,
+}
+impl Meshable for Circle2d {}
+
+/* REFERENCE ONLY
+struct BoundingCircle2d {
+  circle: Circle2d,
+  translation: Vec2,
+}
+impl Meshable for BoundingCircle2d {}
+impl Bounding2d for BoundingCircle2d {}
+
+struct CircleCollider2d {
+  sphere: Circle2d,
+  translation: Vec2,
+}
+impl Meshable for CircleCollider2d {}
+impl Collider2d for CircleCollider2d {}
+*/
+
+// Box Types
+
+struct Box2d {
+  half_extents: Vec2,
+}
+impl Meshable for Box2d
+
+/* REFERENCE ONLY
+struct BoundingBox2d {
+  box: Box2d,
+  translation: Vec2,
+}
+impl Meshable for BoundingBox2d {}
+impl Bounding2d for BoundingBox2d {}
+type Aabb2d = BoundingBox2d;
+
+struct BoxCollider2d {
+  box: Box2d,
+  translation: Vec2,
+  rotation: Mat2,
+}
+impl Meshable for BoxCollider2d {}
+impl Collider2d for BoxCollider2d {}
+type Obb2d = BoxCollider2d;
+*/
+
+// Capsule Types
+
+struct Capsule2d {
```

Comment:
I'm not convinced the `Capsule2d` should be first-class primitive.

Instead it should look like this:
```Rust
struct CapsuleCollider2d {
  capsule: Capsule, // Using the "3d Capsule" as the primitive representation
  translation: Vec2,
  rotation: Mat2,
}
````